File: 0119. Pascal's Triangle II/solution.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        pascal_tri = []
        if rowIndex <= 0:
            pascal_tri = [1]
        else:
            pascal_tri = [1, 1]
            if rowIndex > 1:
                for i in range(2, rowIndex+1):
                    # Calculate the new row by using previous row elements in inverse order
                    for j in range(i-1, 0, -1):
                        pascal_tri[j] = pascal_tri[j] + pascal_tri[j-1]
                    pascal_tri.append(1)
        return pascal_tri

File: 0119. Pascal's Triangle II/solution.java
/**
 * My solution
 */
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> row = new ArrayList<>();
        row.add(1);
        while (rowIndex-- > 0) {
            int rowSize = row.size();
            int prev = rowSize > 1 ? 1 : 0;
            for (int i = 1; i < rowSize; i++) {
                int curr = row.get(i);
                row.set(i, prev + curr);
                prev = curr;
            }
            row.add(1);
        }
        return row;
    }
}
